Add ACL package mode (mint) + package operator mode (transfer/approve/revoke) (#249)
casper-ecosystem/cep-78-enhanced-nft: docs/modalities.md

1. `Unlocked`: The ACL whitelist is unlocked and can be updated via the set variables endpoint.
2. `Locked`: The ACL whitelist is locked and cannot be updated further.

> If the `WhitelistMode` is set to `Locked` an ACL whitelist of entity keys must be provided on installation. This whitelist dictates which entities can mint NFTs in the restricted `ACL` minting mode. These entities include `Accounts`, `Contracts` and `Contracts from a package` with [ ACL_PACKAGE_MODE]().

Reviewer: ```suggestion
If the `WhitelistMode` is set to `Locked` an ACL whitelist of entity keys must be provided on installation. This whitelist dictates which entities can mint NFTs in the restricted `ACL` minting mode. These entities include `Accounts`, `Contracts` and `Contracts from a package` with [ACL_PACKAGE_MODE]().
```

Should there be this space before ACL_PACKAGE_MODE within the brackets?

Author: no but a link to the mode between () actually

Author: Please see this commit thanks

https://github.com/casper-ecosystem/cep-78-enhanced-nft/pull/249/commits/55060ea6b86f7d9eb820a3df914df66858845a13

Author: Removed the space too thanks

| Public      | 1   |
| ACL         | 2   |

## AllowMinting

> By switching allow_minting to false we pause minting. Reverts with MintingIsPaused error if allow_minting is false. This modality provides two options:

Reviewer: ```suggestion
The `AllowMinting` modality dictates the ability for **any** entity to mint or not. If `allow_minting` is set to false, any attempt to mint will revert with a `MintingIsPaused` error. This modality provides two options:
```

| AllowMinting | bool |
| ------------ | ---- |

## ACLPackageMode

> The ACL package mode governs the ability for whitelisting a package instead of a versioned contract. By whitelisting a package, a new contract version will be automaticly whitelisted for minting after its package upgrade. This modality provides two options:

Reviewer: ```suggestion
The ACL package mode modality governs the ability for whitelisting a package instead of a versioned contract. By whitelisting a package, a new contract version will be automatically whitelisted for minting after its package upgrade. This modality provides two options:
```

| ACLPackageMode | bool |
| -------------- | ---- |

> This modality is an optional installation parameter and will default to the `false` if not provided. This

Reviewer: ```suggestion
This modality is an optional installation parameter and will default to `false` if not provided. This mode can be changed on contract upgrade or can be updated via the set variables endpoint.
```

> mode can be changed on contract upgrade or can be updated via the set variables endpoint.

Reviewer: ```suggestion

```

Not sure how this ended up as two lines.

> > Before using this modilty, please understand the security implications of having automatic whitelisting of versioned contracts.

Reviewer: ```suggestion
> Before using this modality, please understand the security implications of having automatic whitelisting of versioned contracts.
```

## PackageOperatorMode

The package operator mode governs the ability for approving a package instead of a versioned contract. By approving a package as an operator, a new contract version will be automaticly approved or approved for all for transfer after its package upgrade. This modality provides two options:

| PackageOperatorMode | bool |
| ------------------- | ---- |

> This modality is an optional installation parameter and will default to the `false` if not provided. This

Reviewer: ```suggestion
This modality is an optional installation parameter and will default to `false` if not provided. This mode can be changed on contract upgrade or can be updated via the set variables endpoint.
```

> mode can be changed on contract upgrade or can be updated via the set variables endpoint.

Reviewer: ```suggestion

```

> > Before using this modilty, please understand the security implications of having automatic approvals and revocation right of versioned contracts.

Reviewer: ```suggestion
> Before using this modality, please understand the security implications of having automatic approvals and revocation rights for versioned contracts.
```
